npm/pkg/controlplane/translation/translatePolicy_test.go

```diff
@@ -367,6 +367,59 @@ func TestIPBlockIPSet(t *testing.T) {
 			},
 			translatedIPSet: ipsets.NewTranslatedIPSet("test-in-ns-default-0IN", ipsets.CIDRBlocks, []string{"172.17.0.0/16"}...),
 		},
+		{
+			name:        "0.0.0.0/0 cidr",
+			ipBlockInfo: createIPBlockInfo("test", "default", policies.Ingress, policies.SrcMatch, 0),
+			ipBlockRule: &networkingv1.IPBlock{
+				CIDR: "0.0.0.0/0",
+			},
+			translatedIPSet: ipsets.NewTranslatedIPSet("test-in-ns-default-0IN", ipsets.CIDRBlocks, []string{"0.0.0.0/1", "128.0.0.0/1"}...),
+		},
+		{
+			name:        "0.0.0.0/0 cidr and one 10.0.0.0/1 except",
+			ipBlockInfo: createIPBlockInfo("test", "default", policies.Ingress, policies.SrcMatch, 0),
+			ipBlockRule: &networkingv1.IPBlock{
+				CIDR:   "0.0.0.0/0",
+				Except: []string{"10.0.0.0/1"},
+			},
+			translatedIPSet: ipsets.NewTranslatedIPSet("test-in-ns-default-0IN", ipsets.CIDRBlocks, []string{"0.0.0.0/1", "128.0.0.0/1", "10.0.0.0/1nomatch"}...),
+		},
+		{
+			name:        "0.0.0.0/0 cidr and one 0.0.0.0/1 except",
+			ipBlockInfo: createIPBlockInfo("test", "default", policies.Ingress, policies.SrcMatch, 0),
+			ipBlockRule: &networkingv1.IPBlock{
+				CIDR:   "0.0.0.0/0",
+				Except: []string{"0.0.0.0/1"},
+			},
+			translatedIPSet: ipsets.NewTranslatedIPSet("test-in-ns-default-0IN", ipsets.CIDRBlocks, []string{"0.0.0.0/1nomatch", "128.0.0.0/1"}...),
+		},
+		{
+			name:        "0.0.0.0/0 cidr and one 128.0.0.0/1 except",
+			ipBlockInfo: createIPBlockInfo("test", "default", policies.Ingress, policies.SrcMatch, 0),
+			ipBlockRule: &networkingv1.IPBlock{
+				CIDR:   "0.0.0.0/0",
+				Except: []string{"128.0.0.0/1"},
+			},
+			translatedIPSet: ipsets.NewTranslatedIPSet("test-in-ns-default-0IN", ipsets.CIDRBlocks, []string{"0.0.0.0/1", "128.0.0.0/1nomatch"}...),
+		},
+		{
+			name:        "0.0.0.0/0 cidr and two 0.0.0.0/1 and 128.0.0.0/1 except",
+			ipBlockInfo: createIPBlockInfo("test", "default", policies.Ingress, policies.SrcMatch, 0),
+			ipBlockRule: &networkingv1.IPBlock{
+				CIDR:   "0.0.0.0/0",
+				Except: []string{"0.0.0.0/1", "128.0.0.0/1"},
+			},
+			translatedIPSet: ipsets.NewTranslatedIPSet("test-in-ns-default-0IN", ipsets.CIDRBlocks, []string{"0.0.0.0/1nomatch", "128.0.0.0/1nomatch"}...),
+		},
+		{
+			name:        "0.0.0.0/0 cidr and duplicated 0.0.0.0/1 and 128.0.0.0/1 except",
+			ipBlockInfo: createIPBlockInfo("test", "default", policies.Ingress, policies.SrcMatch, 0),
+			ipBlockRule: &networkingv1.IPBlock{
+				CIDR:   "0.0.0.0/0",
+				Except: []string{"0.0.0.0/1", "128.0.0.0/1", "128.0.0.0/1"},
```

Comment:
by the test name, did you mean to have two "0.0.0.0/1" in the excepts?

Reply:
Reorganize UTs for better understanding.